aws/cloudformation/standalone/gen_ai_curriculum/deploy_gen_ai_curriculum_stack

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env ruby
+require 'aws-sdk-applicationautoscaling'
+require 'colorize'
+require 'optparse'
+require 'open3'
+require 'pycall'
+require_relative './config'
+
+def main
+  options = {}
+  OptionParser.new do |opts|
+    opts.banner = "Usage: ./deploy_gen_ai_curriculum_stack [options]"
+
+    opts.on("-e", "--environment ENVIRONMENT", String, "Stack environment (prod or test)") do |v|
+      options[:environment] = v
+    end
+
+    opts.on("-d", "--dry-run [FLAG]", TrueClass, "Dry run (create template only but don't deploy)") do |v|
+      options[:dry_run] = v.nil? ? true : v
+    end
+
+    opts.on("-v", "--image-uri-version", Numeric, "HuggingFace Image URI version") do |v|
+      options[:image_uri_version] = v
+    end
+
+    opts.on("-i", "--image-uri IMAGE_URI", String, "HuggingFace Image URI to use") do |v|
+      options[:image_uri] = v
+    end
+  end.parse!
+
+  # Prompt for environment if not provided
+  unless options[:environment]
+    print "Environment (prod or test): "
+    options[:environment] = gets.chomp
+  end
+
+  unless %w[prod test].include?(options[:environment])
+    puts "Error: Invalid environment '#{options[:environment]}'. Must be 'prod' or 'test'."
+    exit 1
+  end
+
+  # Prompt for dry-run if not provided
+  unless options.key?(:dry_run)
+    print "Dry-run (create template only)? (y/n): "
+    response = gets.chomp.downcase
+    options[:dry_run] = response == 'y'
+  end
+
+  unless options.key?(:image_uri)
+    unless options.key?(:image_uri_version)
+      print "HuggingFace Image URI version (blank for latest): "
+      options[:image_uri_version] = gets.chomp
+    end
+    puts "\nFetching HuggingFace Image URI..."
+    sagemaker = PyCall.import_module("sagemaker.huggingface")
+    image_uri = if options[:image_uri_version].empty?
+                  sagemaker.get_huggingface_llm_image_uri("huggingface")
+                else
+                  sagemaker.get_huggingface_llm_image_uri("huggingface", options[:image_uri_version])
+                end
```

Comment:
@snickell being able to do this inline with PyCall is super slick 😎 let me know if this looks right or if there's a better way to do this

Reply:
Oh yeah I think I just have `sagemaker` installed locally via manual `pip` call (technically `python3 -m pip install` I think) 😅 let me try these steps and uninstall my local version and see if this works

Reply:
If for some reason the above process doesn't update the root `/pdm.lock`, would you let me know?

But then you can immediately workaround it (happy to have it merged this way, but rather try the other way first) by: 

1) reverting all your changes to `/python/pycdo/pyproject.toml` and `/python/pycdo/pdm.lock`
2) running `pdm add sagemaker` from the root directory (i.e. NOT in /python/pycdo this time)
3) commit both code-dot-org/pdm.lock and code-dot-org/pyproject.toml

Reply:
Okay interestingly, reverting the `pdm.lock` and `pyproject.toml` changes and re-running `pdm add setuptools` in `python/pycdo` also works and doesn't result in any warnings.

Reply:
After offline discussion with @snickell we decided to remove PyCall here - reason is that the sagemaker dependency is pretty large and we're only using it to compute a static string value (HF image URI). Instead we now hardcode a specific image URI that we know works (using version 2.2.0 which is the latest version as of this comment) and print instructions for how to generate a new image URI. 

Here's that section of the console output (very open to suggestions to make this clearer and easier to follow 🙂)
<img width="999" alt="Screenshot 2024-09-04 at 9 46 31 PM" src="https://github.com/user-attachments/assets/332602f4-7ca2-413d-9017-9d42f34896da">